### ISSUE TITLE
Noise improvements.

### ISSUE THREAD

**Issue by Pmax65** [hugen79/NanoVNA-H#14]
I improved the noise adding few caps to the Silicon Labs clock generator chip and now the standalone instrument is fully working up to 1.5GHz with only a very little residual noise above 1.4GHz that it is appreciable in the Smith chart only. 
I still have some issue with the PC communication because of the battery charger that disturbs the clock. I already thought a solution for this too, but I'm waiting for the sot23 P-MOSfet, to check if it works as thought. Once I fixed it I'll publish the "how to" of the whole hardware fixing.
In the meantime I have a couple of desiderata to your code.
1) the flashing LED 2 is very noisy and useless. It's much better to keep it off, if your intention was to give a pace indicator, I suggest you to implement it as a virtual indicator somewhere on the LCD screen (I currently removed its limiting resistor to keep it off). 
2) Since there is still a little noise above 1.4GHz, It would be great if the calibration was done not by one acquisition of the raw data per calibration step as it is now. It should be better if you get more scans of the raw data and average them to get the correction coefficients averaged on more scans (say 10 scans, for example). This will take a little more time during the single calibration steps, but it  should warrant that the noise is corrected on its averaged value and not randomly on the instantaneous values acquired during the single scan.
Have a great day.
Massimo

**Comment by qrp73** [comment]
regarding to LED 2, NanoVNA-Q disable it during sweep cycle to avoid noise: https://github.com/qrp73/NanoVNA-Q/releases

so, there is no need to desolder something. You can just change firmware.

**Comment by Pmax65** [comment]
Not really so.
The clock generator's inner PLLs are always locked also during the non scanning time. 
So the perturbation on the integrators of those PLLs still deteriorate their output phase noise during the scan time.

BTW, even if I seen that in the latest versions the LED 2 is apparently off during the scan, using a DSO on the limiting resistor pad, I always see some voltage spikes of few ms in duration there.

For these reasons I removed that resistor and the noise above 900MHz dropped a lot, despite I already downloaded the latest version of  NanoVNA-Q.

My current mods consist on:
1) added one 4.7F X7R capacitor in parallel to C9 and one other on C10
2) added one 68uF/6.3 tantalum capacitor in parallel to C9 and the 4.7uF already in parallel as above
3) added one 68uF/6.3 tantalum capacitor soldered on U3 input pin and grounded to the ground plane near Y1
4) removed R3

Doing the above, after a correct calibration, I get a Smith chart plot with just a little noisy plot of 2/3mm above 1.3GHz, making the instrument well usable in the HAM radio 23 cm band for antennas and filters checkings.
The SWR plot at 1.5GHz with the calibrated dummy load on the CH0 port shows 1:1.1 max 1:1.2 spikes, but below 1.3GHz is flat to 1:1
The return loss with the reference load used for the calibration, at 1.5GHz is better than -25dB on any board I checked, not that bad indeed
I did the test on 6 boards of three different versions, two having the ground clips and one having the nanoVNA-H board. And the results are almost the same.
Attaching the USB port, I still have noise above 900MHz, but I already checked that it is due to the battery charging. Detaching the battery positive wire and powering the VNA by the USB only, after a new calibration, the results are the same for the stand-alone running reported above.
For this reason I planned to put a P-MOSfet in series to the battery to detach the battery from the circuitry when the instrument is turned on and the power come from the USB port.
This way the battery charging will be still allowed when the instrument is turned off, but it's thought to remove that noise during the instrument is powered on. I just to check it when the MOSfets will be delivered.

Have a great day.

Massimo

**Comment by qrp73** [comment]
> Not really so.
> The clock generator's inner PLLs are always locked also during the non scanning time.
> So the perturbation on the integrators of those PLLs still deteriorate their output phase noise during the scan time.
> 

No, it performs PLL reset at the beginning of sweep. So, any voltage fluctuations outside sweep time doesn't matter.

NanoVNA-Q has 10 ms stabilization delay after LED off, before it starts sweep.

**Comment by Pmax65** [comment]
At this point I don't really know why I seen that sensible improvement after having removed R3.

Anyways, I'm sure that I also seen some glitches during the sweep. Let me return to home because now I'm in office and I can't check it one more time until I'll be back to home.

**Comment by qrp73** [comment]
You're seen it, because used firmware which uses LED blinking during sweep. NanoVNA-Q firmware disable LED during sweep and don't blink until the end of sweep.

**Comment by Pmax65** [comment]
No, I'm not sure that I seen it with V0.4.3 indeed, but I'm sure that I seen it with your V 0.4.2, and (if I'm not wrong) that function should have been already done. 

One doubt: 
what is your noise above 1.3GHz without any additional caps on the 5V & 3.3V power lines?
My doubt is that having increased the capacitance on the 3.3V U3 output, that could have introduced some sort of instability at the voltage regulator output after current pulses on it's input that lasts more than those 10ms.
I must check resoldering R3.
I'll tell you this night.

**Comment by Pmax65** [comment]
Hi qrp73.
yes you are right and I was wrong I didn't realized that when the LED2 is on, there is  a PWM signal on it and having the horizontal scale of the DSO on 1ms/div I probably confused the ON state with OFF state of the LED2 when the PWM was there.
Here below you can see the 1.3GHz and 1.5GHz with the LED blinking.
(please note that I had to detach the battery + wire to get these plots from the nanoVNA Saver, because I still haven't received the MOSfets).
What do you think about the 10 time averaging suggestion during calibration?
It could better correct the scan above those 1.3GHz.

Thank you for sharing your work about this nice device.

![With LED  1 3GHz](https://user-images.githubusercontent.com/57619411/68613047-2b998480-04be-11ea-9584-78e813d842d6.JPG)
![With LED 1 5GHz](https://user-images.githubusercontent.com/57619411/68613048-2b998480-04be-11ea-8dae-23a232bb7bf9.JPG)

**Comment by hugen79** [comment]
Noise problems introduced by USB connection I will use the new power chip in NanoVNA-H V3.4 to improve this problem. I will try to increase the capacitance of the si5351 to reduce noise. The problem of LED flicker noise can be improved by software. If you prefer, you can send your address to my email at hugen@outlook.com and I will send you the modified PCB. Thank you!

**Comment by Pmax65** [comment]
hi hugen79,
first off all, thank you for considering this issue, and for sharing you work on the nanoVNA-H board.

Since I had 6 nanoVNA on the table (only one was mine, the others were owned by friends of my local radio-club, I'm not a collector :-) of course), I believe I missed to update one and probably I did the check on that one, because I well remember how the noise dropped down after having removed R3.
Anyways, considering how much the instrument is sensitive to the power supply noise I would keep that LED off since in my opinion is useless, but this is just an opinion of course.
About the USB connection I already ordered the MOSFETs and I hope they'll come here soon (Italian Postal Service permitting). I planned to detach the battery when the USB power is applied and the instrument is turned on, it should take just 3 components comprising the MOSFET, not that complicated indeed. I probably mount the components on a very small piece of prototyping board and solder it in place of the battery connector.

Anyways your nanoVNA-H 3.4 PCB is surely well accepted, I 'm curious to know how it is handled the new power chip.

By the way, do you know from where it cames the R40, C45 and C46 network configuration?
I admit that I know very little about the electrical layer USB specs, but do you know what's the usefulness of the 2ohm resistor in series to the second 10uF capacitance?

I'll send you a direct e-mail for the 3.4 PCB.

Have a great day.

Massimo

**Comment by hugen79** [comment]
The specifications of r40, c45, and c46 are from the data sheet of Ip5303. The series r40 and c45 are used to ensure the loop stability of ip5303. However, when an external USB power supply is connected, the MOSFET inside the ip5303 turns on and uses an external USB power supply, causing external interference to enter the power supply. For the new PCB, I will use the FM9688, a power chip widely used in Bluetooth headset chargers in China. The chip charging circuit and DC-DC circuit work independently, and the external circuit is simple and easy to install on the nanoVNA PCB. on.

**Comment by Pmax65** [comment]
Hi hugen79,
thank you for the explanation, I seen that in the datasheet it was there, but (at least in the English version) I didn't find any explanation for it.

>However, when an external USB power supply is connected, the MOSFET inside the ip5303 turns on
>and uses an external USB power supply, causing external interference to enter the power supply.
I'm not sure the noise come from the USB line, I suppose is something due to the pulsed current into the battery during the charging process, because if you remove the battery and leave the USB power alone the noise is no longer there. For this reason I thought to put a MOSFET based switch between the battery and the Ip5303.
About the FM9688: I haven't found the english datasheet. I'm no way able to read the Chinese ideograms :-(
Have a great day.
Massimo

**Comment by qrp73** [comment]
> causing external interference to enter the power supply.

According to my tests, this is not external interference. You can apply very clean external power (with no noise), but anyway it will be noisy. Also it is not related with cable, because there is no noise when usb cable is connected, but external voltage is not supplied.

I think the root of cause may be the following:
- IP5303 generate noise
- external voltage leads to a little different power supply voltage and it leads to some change in behavior

**Comment by hugen79** [comment]
> > causing external interference to enter the power supply.
> 
> According to my tests, this is not external interference. You can apply very clean external power (with no noise), but anyway it will be noisy. Also it is not related with cable, because there is no noise when usb cable is connected, but external voltage is not supplied.
> 
> I think the root of cause may be the following:
> 
> * IP5303 generate noise
> * external voltage leads to a little different power supply voltage and it leads to some change in behavior

The difference in voltage will cause the gain of the sa612 to differ, resulting in errors. Maybe a separate charge pump can further reduce the noise, and I will do some tests.

**Comment by Pmax65** [comment]
I perfectly agree with qrp73 when he wrote:

> According to my tests, this is not external interference. You can apply very clean external power (with no noise), but anyway it will be noisy. Also it is not related with cable, because there is no noise when usb cable is connected, but external voltage is not supplied.

In my opinion the problem is in the charging current pulses coming from the USB supply that produces inevitable little voltage spikes on the battery, since the voltage of the battery influence the calibration, in my opinion it's clear that those spikes are the cause of the performance deterioration.

I don't know if FM9688 will fix the issue, for doing that it should have two different switching mode PSU inside: one step-up for driving the load and one step-down for charging the battery; they should have very different PWM frequency to allow the step-up to be more reactive to the battery voltage changing caused by the lower frequency step-down charger.
In the datasheet, I see the schematic of the FM9688  and I don't see other than one inductor in that design, this lead me to think that it should not be much better than the Ip5303 under this context, but I could be wrong, of course.

Have a great day.

Massimo

**Comment by hugen79** [comment]
I tried to replace the XC6206 with an ultra-low noise LDO RT9193, and I found that the noise has dropped. I will make a PCB test.
hugen

**Comment by Pmax65** [comment]
> I tried to replace the XC6206 with an ultra-low noise LDO RT9193, and I found that the noise has dropped. I will make a PCB test.
Nice one. If it's confirmed the "thorn in the side" of the instrument is the clock generator, since the mixers are not powered from the 3.3V regulator.

**Comment by hugen79** [comment]
> > I tried to replace the XC6206 with an ultra-low noise LDO RT9193, and I found that the noise has dropped. I will make a PCB test.
> 
> Nice one. If it's confirmed the "thorn in the side" of the instrument is the clock generator, since the mixers are not powered from the 3.3V regulator.


Sorry, I made a mistake. Due to my wrong connection to the RT9193 pin, the actual output voltage of the LDO rises to 3.6V, and the performance improvement is due to the voltage rise. But this gave me a new idea to improve performance, I am making new attempts.

**Comment by d51r3verse** [comment]
https://github.com/qrp73/NanoVNA-Q/issues/13#issuecomment-552679259
Mine(Green PCB) was around 6.5 MHz

**Comment by hugen79** [comment]
> [qrp73/NanoVNA-Q#13 (comment)](https://github.com/qrp73/NanoVNA-Q/issues/13#issuecomment-552679259)
> Mine(Green PCB) was around 6.5 MHz

Thanks for your feedback, I will modify the schematic in v3.4.

**Comment by Pmax65** [comment]
Hi hugen79,
> Sorry, I made a mistake. Due to my wrong connection to the RT9193 pin, the actual output voltage of the LDO rises to 3.6V, and the performance improvement is due to the voltage rise. But this gave me a new idea to improve performance, I am making new attempts.

Don't worry:
anyone who has never made a mistake has never tried anything new!
(Albert Einstein) 

:-) 

I would have here the mosfets and tell you if the patch works for the USB powering. 
Anyways, as I have them here, I'll surely inform you about how the patch works.

For the 7.5/6.5Mhz peak issue: 
I didn't noticed of it, but I must say that I have already added the 100nF on the mixer as per the NXP datasheet. Since I don't see anything there on any of the 3 different versions I have here, I can confirm you that it works fine.
I didn't reported that caps mod, just because I was convinced that they did nothing to the instrument, since I never looked to that narrow range before.
My compliment to Owen for having catched that issue. Here in Italy we say: "he catched the the needle in the haystack".

**Comment by coralenka** [comment]
Hi
![image](https://user-images.githubusercontent.com/45734739/68790785-a397b400-0650-11ea-8f0b-df886af3842b.png)

**Comment by Pmax65** [comment]
Hi coralenka
Fantastic plots, how do you get them so noise-free up to 1.5GHz?

In the meantime, I just received the MOSFETs and checked the USB switcher and it works good, but above 1.3GHz as per the battery powered mode the noise is still there.

The mod consists of place on a small piece of proto-board (3 by 2 pads) an AO3401 P-channel MOSFET as follows:
DRAIN connected to the positive wire of the battery 
SOURCE connected to the PCB positive battery pad
GATE connected to the ANODE of one 1N4148W diode and a 10kohm resistor (both on the little piece of proto-board).
The diode CATHODE is connected through a short wire to U3pin3, while the other pad of the 10kohm resistor is connected through a short wire to U2pin1.
Doing that circuitry, the battery charges only when the nanoVNA is not powered but the USB plug is connected to a current source.
When the nanoVNA is powered on, the battery is disconnected and the noise is as in the plots attached below.
Have a great day.

Massimo
![1 5GHz with MOSFET switch](https://user-images.githubusercontent.com/57619411/68810612-e9fc0b80-066e-11ea-88f3-abc15f0cdd02.JPG)
![1 3GHz with MOSFET switch](https://user-images.githubusercontent.com/57619411/68810613-e9fc0b80-066e-11ea-8152-d1b893486823.JPG)

**Comment by coralenka** [comment]
HiI made no mod to my device yet

Trimis din Yahoo Mail pe Android 
 
  Pe joi, nov. 14, 2019 la 0:45, Pmax65<notifications@github.com> a scris:   
Hi coralenka
Fantastic plots, how do you get them so noise-free up to 1.5GHz?

In the meantime, I just received the MOSFETs and checked the USB switcher and it works good, but above 1.3GHz as per the battery powered mode the noise is still there.

The mod consists of place on a small piece of proto-board (3 by 2 pads) an AO3401 P-channel MOSFET as follows:
DRAIN connected to the positive wire of the battery
SOURCE connected to the PCB positive battery pad
GATE connected to the ANODE of one 1N4148W diode and a 10kohm resistor (both on the little piece of proto-board).
The diode CATHODE is connected through a short wire to U3pin3, while the other pad of the 10kohm resistor is connected through a short wire to U2pin1.
Doing that circuitry, the battery charges only when the nanoVNA is not powered but the USB plug is connected to a current source.
When the nanoVNA is powered on, the battery is disconnected and the noise is as in the plots attached below.
Have a great day.

Massimo

**Comment by Pmax65** [comment]
> I made no mod to my device yet

Hi coralenka,
interesting. 
Has your nanoVNA the battery installed?
I ask because when attached to the USB port, I experienced a great increase in noise bove 900MHz in all the 3 hardware versions I checked.
Your seems incredibly quiet even when attached to the PC.
Have a great day.
Massimo

**Comment by coralenka** [comment]
Yes the battery is attached and at the same time the device is connected to the computer...I use the hardware version 0.4.0-273 de YO4AUL 

Trimis din Yahoo Mail pe Android 
 
  Pe joi, nov. 14, 2019 la 15:14, Pmax65<notifications@github.com> a scris:   

I made no mod to my device yet


Hi coralenka,
interesting.
Has your nanoVNA the battery installed?
I ask because when attached to the USB port, I experienced a great increase in noise bove 900MHz in all the 3 hardware versions I checked.
Your seems incredibly quiet even when attached to the PC.
Have a great day.
Massimo

**Comment by Pmax65** [comment]
Hi coralenka,,
it's gorgeous.
Could you publish a photo of the nanoVNA?
Have a great day.
Massimo

**Comment by coralenka** [comment]
This is my device


Trimis din Yahoo Mail pe Android 
 
  Pe joi, nov. 14, 2019 la 20:12, Pmax65<notifications@github.com> a scris:   
Hi coralenka,,
it's gorgeous.
Could you publish a photo of the nanoVNA?
Have a great day.
Massimo

**Comment by Pmax65** [comment]
HI coralenka,
I'm not sure, did you send anything about how it looks your nanoVNA?
I can't see anything.

**Comment by coralenka** [comment]
This is my device  

Trimis din Yahoo Mail pe Android 
 
  Pe vin., nov. 15, 2019 la 1:02, Pmax65<notifications@github.com> a scris:   
HI coralenka,
I'm not sure, did you send anything about how it looks your nanoVNA?
I can't see anything.

**Comment by Pmax65** [comment]
It seems there is something wrong in the file handling because I can't see anything.
Could you send the photo to my e-mail?
My address is mporzio@teletu.it
I would like to know what HW it is, because I've some nanoVNAs from friends of my local radio club to modify and above 1.3GHz it seems really hard task to comply.
Thank you for considering my requests.
Have a nice day.
Massimo - IK1IZA

**Comment by hugen79** [comment]
The new pcb seems to be better.
![TIM截图20191115221257](https://user-images.githubusercontent.com/32199654/68949621-6ae50f80-07f5-11ea-9102-41a1391182bf.png)

**Comment by hugen79** [comment]
After the port is calibrated, the 20cm RG316 cable is measured, and it seems that 1.3GHz can work normally.
![TIM截图20191115222444](https://user-images.githubusercontent.com/32199654/68950531-4db14080-07f7-11ea-8730-ad6b9f001b5f.png)

**Comment by Pmax65** [comment]
Hi hugen79,
this is more or less what I got with the one I have at home and I think that it's a nice tool for HAM radio requirements. 
How do you dealt with the USB charging noise?
Have a great day.

Massimo

**Comment by Pmax65** [comment]
Hi hugen79,
maybe I've been not clear in my previous comment about your new pcb, I was referring to the noise of course. 
About the accuracy and precision of the measurement, it seems that the older ones have a sort of resonance around 1180MHz that (despite the noise is quite low), it practically invalidate any useful Z measurement up there, but most of the HAM radio would use this instrument for antenna measurements which which have not such precision and accuracy requirements, so I guess it's good enough for them. In my case, who I own an HP8711A VNA (yes it was scalar, but some years ago I hacked it firmware to turn it into an HP8713A which is vectorial), I think this instrument is still a great deal for on field antenna or filters measurements.

You new PCB seems perform much better instead.
Have a great day.

Massimo

**Comment by hugen79** [comment]
> Hi hugen79,
> this is more or less what I got with the one I have at home and I think that that make it a nice tool for HAM radio requirements.
> How do you dealt with the USB charging noise?
> Have a great day.
> 
> Massimo

The pcb using the FM9688 seems to have a smaller usb noise, and a pcb with tantalum capacitors is being produced. I will test it further.、

hugen

**Comment by Pmax65** [comment]
> The pcb using the FM9688 seems to have a smaller usb noise, and a pcb with tantalum capacitors is being produced. I will test it further.、

hugen

Nice one.

**Comment by Pmax65** [comment]
HI coralenka,
I tried to reply to your email directly, but I received a deny alert from my mail server.
I was asking you a photo of the components side of your nanoVNA and (if possible, I don't want bore you) three photos of the instrument while it is measuring the open, short and load references.

Thank you for considering my requests.

Have a great day.

Massimo

**Comment by d51r3verse** [comment]
Found another glitches around at CENTER 64 MHz / SPAN 80 MHz ( Not connected anything / Uncalibrated sweep )
Maybe influenced by 6.5 MHz issue?
But, this sweep does not glitches CH0 Delay.

**Comment by hugen79** [comment]
> Found another glitches around at CENTER 64 MHz / SPAN 80 MHz ( Not connected anything / Uncalibrated sweep )
> Maybe influenced by 6.5 MHz issue?
> But, this sweep does not glitches CH0 Delay.

Is it appearing at 100MHz?

**Comment by Pmax65** [comment]
Hi hugen79,
I finally gave a look to the reflectometric bridge section and found some weird things.
I refer to your schematic Rev. 3.0 which seems reflect the components mounted on the nanoVNAs I've here on my desk.
I can't understand the usefulness of R13. That resistor placed there almost completely invalidates the function of the reference mixer U6 because the changes at the input port of the bridge when the DUT has very different Z are practically no longer measured because of that.
I made R13 = 0ohm and now everything works great till 1500MHz.
Now, without that attenuation resistor, when an open reference is attached to the DUT port of the bridge, the U7 single ended input at pin IN-B exceeds the mixer 3rd order intercept point specs of -13dBm, but since that limit is determined by the conversion gain, it should not be a real issue since the differential input in that case is limited to -19dBm indeed.
I also changed R9 from 82ohm to 150ohm to better match the DUT port Z to 50ohm (it seems that the previous value didn't take account of the Si5351A-B-GT output impedance which is 85ohm).
I applied the mods only on one of the nanoVNAs that I've and not the measurements looks much better.
I also removed R29, R30 and R31 to give "a push" to the LO signal. I know that the SA612AD suggest to center the value between 200  and 300mVpp, but I suppose they do that to avoid spurious harmonics on the LO signal to drive the mixer, but we need that instead for the higher bands that use the odd harmonics method.
The attached images show the results with this configuration.
Note that the shield were not there because the nanoVNA arrived without it and I'm still waiting for the ground clips.
The mounted mixer are SA602A not SA612A, but that seems not critical.

Have a great day,

Massimo - IK1IZA

![Load](https://user-images.githubusercontent.com/57619411/69100374-55721e80-0a5d-11ea-864b-f4cc58057193.JPG)
![Open](https://user-images.githubusercontent.com/57619411/69100376-560ab500-0a5d-11ea-85f8-4036b708f1b8.JPG)
![Open Cable](https://user-images.githubusercontent.com/57619411/69100377-560ab500-0a5d-11ea-9f67-7a7d8a38b5cb.JPG)
![Short Cable](https://user-images.githubusercontent.com/57619411/69100378-560ab500-0a5d-11ea-8d84-55f6649d61e0.JPG)
![Short](https://user-images.githubusercontent.com/57619411/69100379-560ab500-0a5d-11ea-8322-60917bc47f3b.JPG)

**Comment by Pmax65** [comment]
After the mods described above, the corrective function of U6 now works, and it highlighted now a software issue at the band switch point of 300 and 900MHz. If the calibration has been made with a large span (for example at full 1.5GHz span) zooming around those points is evident that the calibration correction fails because interpolates between two point having very different correction coefficients (one below the band switch and one above it).
The right algorithm should extend the last calibrated point correction value below up to the band switch point, and should extend the first calibrated point correction value above down to the band switch point. 
A much better algorithm should use the last two calibrated point correction values below and the first two calibrated points correction values above to compute the correction coefficients extending the linear interpolation between them to the two segments of data close to the band switch point that have no correction values computed (it should be an extension of the two interpolation lines).

Note that the glitch is already easily removed calibrating the instrument with the narrow span.

Have a great day.

Massimo IK1IZA

**Comment by hugen79** [comment]
> After the mods described above, the corrective function of U6 now works, and it highlighted now a software issue at the band switch point of 300 and 900MHz. If the calibration has been made with a large span (for example at full 1.5GHz span) zooming around those points is evident that the calibration correction fails because interpolates between two point having very different correction coefficients (one below the band switch and one above it).
> The right algorithm should extend the last calibrated point correction value below up to the band switch point, and should extend the first calibrated point correction value above down to the band switch point.
> A much better algorithm should use the last two calibrated point correction values below and the first two calibrated points correction values above to compute the correction coefficients extending the linear interpolation between them to the two segments of data close to the band switch point that have no correction values computed (it should be an extension of the two interpolation lines).
> 
> Note that the glitch is already easily removed calibrating the instrument with the narrow span.
> 
> Have a great day.
> 
> Massimo IK1IZA

Thanks for your exploration, I conducted simulations and tests, and the results show that the new bridge is more reasonable. I will discuss the modification of the bridge with edy555. The discontinuity of the band switching point can be improved by modifying the codec's PGA.

hugen

**Comment by qrp73** [comment]
> I can't understand the usefulness of R13.

R13 is needed to get 50 ohm output impedance on the port CH0. If you remove it, your CH0 output impedance will be changed and uncertainty of measurement will be increased.

Did you measured output impedace of port CH0? You can do it by measure output amplitude for different load resistance.

**Comment by Pmax65** [comment]
> R13 is needed to get 50 ohm output impedance on the port CH0. If you remove it, your CH0 output impedance will be changed and uncertainty of measurement will be increased.

I simulated the front-end with Multisim 11 and now with R9 moved to 150ohm the device shows 50.3ohm at the DUT port of the bridge.
I also previously measured the CH0 Z with my HP8711A VNA and for that reason I decided to discuss the reliability of the front-end.
If U6 get the value there and not before R13, the bridge input impedance is not critical for the measurements.
The previous DUT impedance was set to 55ohm instead of the 49.6ohm probably expected. I suppose that probably the designer  did not accounted for the output impedance of the Si5351A-B-GT clock generator which is 85ohm not 0ohm (it's just a guess of course).
Now my HP8711A reported a dynamic CH0 Zin of 50.3ohm (the static impedance is higher of course but not pertinent to our context).
Now (up to 1.3GHz) the nanoVNA gave almost the same Smith plot of my HP8711A.
I can't measure the phase/amplitude above 1.3GHz using the HP8711A just because it stops at 1.3GHz, but I suspect that nanoVNA is not performing well up to 1.5GHz, because the ondulation of its return loss plot with the 30cm cable on CH0 port makes me think that there is a resonating element somewhere in the bridge circuitry that before this mod was at abt 1180MHz and now seems have shifted to a little more than 1.5GHz.

By the way, when I wrote that message last night it was almost 1AM here i Italy and I did a mistake, the power values I simulated were peak-to-peak values not RMS, for this reason all my reasonings about the 3rd intercept point of the mixers were probably wrong. The power values should be safely below the specs limits instead.

Have a great day.

Massimo.

**Comment by hugen79** [comment]
Hi Pmax65 , the band switching point discontinuity I have modified the software to get improved, when using harmonics, the modified circuit will output a larger signal, we need to modify the CH1 input appropriately, I modify R22 and R23 to 43 ohms and the R25 is modified to 13 ohms. It seems to work better now. Maybe the U8 input input attenuation should be a little bigger.

hugen

**Comment by hugen79** [comment]
> > R13 is needed to get 50 ohm output impedance on the port CH0. If you remove it, your CH0 output impedance will be changed and uncertainty of measurement will be increased.
> 
> I simulated the front-end with Multisim 11 and now with R9 moved to 150ohm the device shows 50.3ohm at the DUT port of the bridge.
> I also previously measured the CH0 Z with my HP8711A VNA and for that reason I decided to discuss the reliability of the front-end.
> If U6 gets the power value at the bridge input impedance and not before R13, the bridge input impedance shouldn't be critical for the measurements.
> The previous DUT impedance was set to 55ohm instead of the 49.6ohm probably expected. I suppose that probably the designer did not accounted for the output impedance of the Si5351A-B-GT clock generator which is 85ohm not 0ohm (it's just a guess of course).
> Now my HP8711A reports a dynamic CH0 Zin of 50.3ohm (the static impedance is higher of course but not pertinent to our context).
> Now (up to 1.3GHz) the nanoVNA gave almost the same Smith plot of my HP8711A.
> I can't measure the phase/amplitude above 1.3GHz using the HP8711A just because it stops at 1.3GHz, but I suspect that nanoVNA is not performing well up to 1.5GHz, because the ondulation of its return loss plot with the 30cm cable on CH0 port makes me think that there is a resonating element somewhere in the bridge circuitry that before this mod was at abt 1180MHz and now seems have shifted to a little more than 1.5GHz.
> 
> By the way, when I wrote that message last night it was almost 1AM here i Italy and I did a mistake, the power values I simulated were peak-to-peak values not RMS, for this reason all my reasonings about the 3rd intercept point of the mixers were probably wrong. The power values should be safely below the specs limits instead.
> 
> Have a great day.
> 
> Massimo.

I am very sorry, Pmax65, I mistakenly deleted your reply.
You can use the firmware of edy555 to test, the discontinuity will be much smaller, you find serious discontinuity in using the firmware of me and qrp73 because we set different PGA values for the nonlinear state of the previous circuit U6, you improve The circuit has been modified to work with U6. You can use the Edy555 firmware without setting a separate PGA value for the U6 output (IN2 of tlv320aic3204).

**Comment by qrp73** [comment]
> and now with R9 moved to 150ohm the device shows 50.3ohm at the DUT port of the bridge.

you cannot measure output impedance of CH0 with NanoVNA. The only way is to use external RF voltmeter and two loads with non-inductive resistors.

**Comment by qrp73** [comment]
> output impedance of the Si5351A-B-GT clock generator which is 85ohm

As I know, some documentation of Si5351 mentioned 50 ohm impedance of si5351, but according to my measurements, it is a little different. May be 60 ohm or something like that, I didn't tested it deep. And you can change output impedance of si5351 by selecting output current in control register.

**Comment by qrp73** [comment]
> Now my HP8711A reports a dynamic CH0 Zin of 50.3ohm (the static impedance is higher of course but not pertinent to our context).

How did you measured output impedance of CH0 with VNA? As I know this is impossible to measure output impedance with VNA.
Measurement of input impedance for CH0 is even more complicated, because it always has output signal, so you cannot measure it with VNA.

**Comment by Pmax65** [comment]
Hi hugen,

> I am very sorry, Pmax65, I mistakenly deleted your reply.

don't worry it was just a compliment for you, Edy555 and qrt73, that I wrote to thanks you all for your care about this nice instrument.

> You can use the firmware of edy555 to test, the discontinuity will be much smaller, you find serious discontinuity in using the firmware of me and qrp73 because we set different PGA values for the nonlinear state of the previous circuit U6, you improve

As said: anyone who has never made a mistake has never tried anything new! 

Instead, I honestly I was dubious about your:
> the modified circuit will output a larger signal, we need to modify the CH1 input appropriately, I modify R22 and R23 to 43 ohms and the R25 is modified to 13 ohms. It seems to work better now. Maybe the U8 input input attenuation should be a little bigger.

This night I'll check, because I did some measurement of the output power from CH0 and (if I didn't made a mistake, because it was late at night) and I measured more or less the same levels declared for the original nanoVNA specs.
If I remember well, I measured -9dBm @ 1MHz almost flat up to 280MHz and a worst case of -19/-20dBm @ 1.5GHz (narrow band measurements made with an Advantest R3131 spectrum analyzer).

Have a great day.

Massimo

**Comment by qrp73** [comment]
As I know, the only way to measure output impedance is to use RF voltmeter and different loads. 
I'm wonder how did you used VNA to measure output impedance?

**Comment by Pmax65** [comment]
Hi qrp73,

> As I know, some documentation of Si5351 mentioned 50 ohm impedance of si5351, but according to my measurements, it is a little different. May be 60 ohm or something like that, I didn't tested it deep. And you can change output impedance of si5351 by selecting output current in control register.

AFIK, si5351 is slightly different from Si5351A-B-GT. The on-line data sheets differs exactly per their output impedance, the later have improved the output driver for the 8mA current availability and moved the Zout from 50 to 85ohm.

> How did you measured output impedance of CH0 with VNA? As I know this is impossible to measure output impedance with VNA.

Measurement of input impedance for CH0 is even more complicated, because it always has output signal, so you cannot measure it with VNA.

You can do it because the scanned output signal of the DUT is not in sync with the scanned test signal of the VNA, so adjusting the right way the VNA sweep time to get just sporadic dips power for any VNA scan, if you set the VNA to average more scans those sporadic dips are averaged on the whole VNA scan points and their influence become negligible.
The fact that I measured the Multisim 11 predicted value with few tenth of ohm of difference demonstrates it.
Note that I turned off the nanoVNA and I measured exactly the static Z as predicted by Multisim 11.
I'm quiet confident in that measurement I did.

**Comment by qrp73** [comment]
> AFIK, si5351 is slightly different from Si5351A-B-GT

I see that si5351-B (which is used in NanoVNA), according to the datasheet has 50 ohm:
![si5351-impedance](https://user-images.githubusercontent.com/46676744/69140802-ed0e5600-0acb-11ea-8858-8f6835413fbf.png)

> Note that I turned off the nanoVNA and I measured exactly the static Z

I think this is incorrect measurement, because when you turned off NanoVNA, it's ports impedance are changed.

**Comment by d51r3verse** [comment]
> > Found another glitches around at CENTER 64 MHz / SPAN 80 MHz ( Not connected anything / Uncalibrated sweep )
> > Maybe influenced by 6.5 MHz issue?
> > But, this sweep does not glitches CH0 Delay.
> 
> Is it appearing at 100MHz?

I can not see any glitches around 100 MHz. only appeared 6.5 MHz and 64 MHz
![6 5M_power+1(0 004656872)](https://user-images.githubusercontent.com/29874379/69141209-958ddb80-0b07-11ea-8ef2-c6a75657a338.png)
![64M_power+1(0 110340707)](https://user-images.githubusercontent.com/29874379/69141263-b8b88b00-0b07-11ea-8a1f-ecc0903f2731.png)

**Comment by Pmax65** [comment]
> I think this is incorrect measurement, because when you turned off NanoVNA, it's ports impedance are changed.

Maybe I've been not clear. 
I just did that measurement to check the reliability of the Multisim 11 simulator for the static Z computation. 
And it demonstrated to be reliable indeed.

Then I used it to compute the dynamic Z including in the simulation the Si5351A-B-GT Zout, and checked it with VNA and reported almost the same value (few tenths of ohms of difference).
I took an another look to the datasheets, and indeed it seems that the last one (rev 1.1) reports 50ohm (the one reporting 85ohm was revision 0.95). This could be the reason of those few tenths of ohms of difference I measured.
This night I'll check with the simulator for the Si5351A-B-GT Zout 50ohm impedance and I report you the results.

Have a great day.

Massimo

**Comment by qrp73** [comment]
My nanovna also don't have glitch on 100 MHz, but some small glitch is present on 6.5 MHz.

Scale is 0.1 dB:
![glitch-6500kHz](https://user-images.githubusercontent.com/46676744/69142039-a53cfe00-0ace-11ea-8160-e6624a7c923e.png)

**Comment by Pmax65** [comment]
qrp73,
Maybe I missed something in your messages:
> You're talking about input CH0 impedance. But how did you measured output CH0 impedance?

What do you mean with Zin and Zout?
A port has one impedance, I wrote "input" just for indicate the CH0 input port, but there is not a Zin and a Zout. 
Do I missed something?

Massimo

**Comment by qrp73** [comment]
when you use port as a source, this source has some internal impedance, this is output impedance of the port.

When you use port as a load, this load has some internal impedance, this is input impedance of the port.

Input and output impedances are diffrent. 

The port CH0 working simultaneously as a source and as a load. So, both impedances are involved.

**Comment by hugen79** [comment]
> My nanovna also don't have glitch on 100 MHz, but some small glitch is present on 6.5 MHz.
> 
> Scale is 0.1 dB:
> ![glitch-6500kHz](https://user-images.githubusercontent.com/46676744/69142039-a53cfe00-0ace-11ea-8160-e6624a7c923e.png)

The glitch near 100MHz comes from the firmware of edy555. The wait time after switching the division factor of si5351 is not enough. The problem is not existed in nanoVNA-Q.

hugen

**Comment by Pmax65** [comment]
Hi qrp73,

> when you use port as a source, this source has some internal impedance, this is output impedance of the port.
When you use port as a load, this load has some internal impedance, this is input impedance of the port.
Input and output impedances are diffrent, because source used in NanoVNA is non-linear component (si5351), and even more, source impedance depends on it's internal frequency in different way than input impedance depends on external frequency.

Ok, now I get your point, but I can't imagine that an impedance between two points could be different depending on where the signal is coming. 
Could you make me an example where you can demonstrate that an external signal generator with a given Z produces different results (V and I, the phase between them shouldn't matter because is compensated by the reference detector) compared to the same results measured in that very same port attaching an external load of the same Z of that generator?
It shouldn't matter if the generators are not linear because both are monitored and corrected for their effective power signal amplitude and phase to linearize them.
At this point the superimposition of the effects should apply, and the si5351 internal ideal current generator should be considered an open circuit when the external generator is taken in account, so on the CH0 port the impedance should be the same. The same should apply vice-versa when the external generator is considered only for it's Z. 
The only problem should raise when both the signal generators are on the very same frequency, because the reference external VNA could receive the inner generator power vectorially added to its back reflected signal. But as said, making the scan speed of the two sweeps very different the coincidence happens for very little time respect the whole span making the error negligible (not zero of course).

Have a great day.

Massimo

**Comment by qrp73** [comment]
> Could you make me an example where you can demonstrate that an external signal generator with a given Z produces different results (V and I

![zinout-impedance](https://user-images.githubusercontent.com/46676744/69154172-2e146380-0ae8-11ea-8c0c-fc779177079a.png)

**Comment by Pmax65** [comment]
Hi qrp73,
I don't agree that this is the case we are arguing about because:
A) The internal source must drive the two nodes where R3 is joined (the port), so that diode must let the RF pass through it, and if it does the DUT signal pass through too in the opposite direction.
B) the signal is monitored for its amplitude by U6 which must measure phase and amplitude at the bridge input to correct them and get the right value for the DUT reflected power measured at its balanced output.
C) I don't know why you asserts that the Si5351 changes it's output impedance. Being a clock driver I can imagine it's a push-pull mosfets based drive thought to warrant the CMOS output voltage span, with an Rds_on trimmed to get the specified Zout, (otherwise why do they specify it?). You can control the output current, but this doesn't mean that the impedance changes. 
In fact in datasheet Rev.1.1 Page 23 Fig.18 they specify to use a Zo = 50ohm for the micro stripline to drive the devices to avoid high VSWR and consequent echoes.

And again the port joints doesn't know if the signal incoming from the DUT is a reflection of its impedance (Zdut) or it is self generated by the DUT being an external generator. The bridge measures both the same ways. To avoid the unwanted signal eventually coming from the DUT it suffices to stay away from its generated frequency, so the unwanted signal doesn't pass through the VNA IF filter.

By the way, maybe that I've been not clear. The Zo that I measured was the one at the lower frequencies (that is, where it has a meaning to be measured). It's obvious that in the Smith chart it wasn't a point but a spiral around the 50ohm value very close to the center of the chart, due to the parasitic components of the bridge network. 
Those ondulation into the S11 return loss graphic visible at the higher frequencies are probably due to that parasitic components, but having reduced the bridge input impedance (making that R13=0ohm) they moved to higher frequency than before.

Have a great day.

Massimo

**Comment by qrp73** [comment]
> I don't know why you asserts that the Si5351 changes it's output impedance

Because there is a code in firmware. The firmware set about 50 Ohm output when frequency is higher than harmonic threshold (300 MHz). When frequency is lower than 300 MHz firmware set some higher impedance, 200 Ohm or something like that.

Could you check input impedance for CH0 for two cases?
1) for sweep 10 MHz - 20 MHz
2) for sweep 310 MHz - 320 MHz

**Comment by Pmax65** [comment]
Hi qrp73,
ok, 
Let me have dinner and this night I'll do it and I tell you the results.

Have a great day (or night? I don't know where you are from so it could be that it's already night there).

Massimo

**Comment by Pmax65** [comment]
Hi qrp73,
here below you can see the plots of the return loss and the Smith charts that I measured on the modified nanoVNA (adjusted for the 50ohm Si5351 Zout).
The mod is now turned as follows to get a low frequency CH0 Zin of about 50.3ohm:
R13 = 0ohm
R9 = 200ohm
Removed R29, R30 and R31

I did the measurement with the nanoVNA emitting a CW signal at 500MHz (upper graphs) and at 100MHz (lower graphs).
Effectively there is a clear change in impedance when in the fundamental band. The RL raise to -20.23dB in the worst case which equates to an uncertainty of about +/- 0.9dB. While in odd harmonics mode the -22.52dB equates to an uncertainty of about +/- 0.6dB.
I don't believe that's so bad, but I don't understand why this switch of impedances is due,
Shouldn't that impedance be left to 50ohm along the entire full 1500MHz span?

Have a great day.

Massimo



![nanoVNA mod](https://user-images.githubusercontent.com/57619411/69198095-e0bde380-0b33-11ea-902a-8fa819ebd989.jpg)

**Comment by hugen79** [comment]
> Hi qrp73,
> here below you can see the plots of the return loss and the Smith charts that I measured on the modified nanoVNA (adjusted for the 50ohm Si5351 Zout).
> The mod is now turned as follows to get a low frequency CH0 Zin of about 50.3ohm:
> R13 = 0ohm
> R9 = 200ohm
> Removed R29, R30 and R31
> 
> I did the measurement with the nanoVNA emitting a CW signal at 500MHz (upper graphs) and at 100MHz (lower graphs).
> Effectively there is a clear change in impedance when in the fundamental band. The RL raise to -20.23dB in the worst case which equates to an uncertainty of about +/- 0.9dB. While in odd harmonics mode the -22.52dB equates to an uncertainty of about +/- 0.6dB.
> I don't believe that's so bad, but I don't understand why this switch of impedances is due,
> Shouldn't that impedance be left to 50ohm along the entire full 1500MHz span?
> In that case up to 700MHz the RL is better than -30dB which equates to gorgeous uncertainty of about +/- 0.28dB.
> 
> BTW, note those small peaks at 200 and 300MHz due to the nanoVNA 100MHz emission, as said they don't influence the impedance analysis on other frequencies, hey just beat .
> 
> Have a great day.
> 
> Massimo
> 
> ![nanoVNA mod](https://user-images.githubusercontent.com/57619411/69198095-e0bde380-0b33-11ea-902a-8fa819ebd989.jpg)

Hi Pmax65，
In the original design of edy555, NanoVNA is only used for 50K-300MHz, so that the si5351 output smaller current can reduce harmonic interference and save current consumption. When I try to use harmonic measurements, I need to get larger harmonics, so I need to increase the output current of the si5351, which will cause the output impedance to decrease.

hugen

**Comment by Pmax65** [comment]
Hi hugen79,
unluckily yesterday night have been busy with the qrp73 impedance change issue and I had not time to check the edy555's cross band correction interpolation fix, and this night I'll be busy with the HF transceiver of an HAM radio friend who stopped transmitting yesterday and became a "red emergency" issue (there are some HAM radio guys like that who are retired and without their RTX are in panic), so I have to suspend my work on this instrument.

I know that edy555 did it for the 50kHz to 300MHz band and I suspected that he used that low current for extend the battery autonomy, I just did not imagine that the Si5351 changed its Zout for doing that, because in digital systems using the wrong impedance for distributing a clock deteriorate the clock signal. I must say that that clock generator was designed prevalently to get fixed frequency clock outputs, so the PCB traces could be designed for the low or high impedance when needed. But I didn't find any mention of that in the datasheet.

Anyways when I wrote the message to qrp73 it was 1:30 and I was not that "awake", so I overestimated the problem because those two plots weren't show the real CH0 Z. That because thinking a little more about it. This morning I realized that the real Z plot is the mix of the twos LOG plots indeed. That is, below 300MHz the Z is the one represented by the lower plot and above 300MHz it is the one represented by the upper plot. So the situation is better than I thought, the uncertainty shouldn't be worse than +/- 0.6dB.
I still never checked the current consumption, but if it was not a problem I would you fix that impedance to 50ohm all along the full span, so the instrument became really reliable below 700MHz.

If you are planning to redo the PCB, I suggest you to reconsider the traces width for their impedance. In the past I worked for 6 years for a safety grade level N+1 redundant radio control factory as designer. In that experience I fought against the PCB trace design for the above 1GHz models. Getting from that experience, I think that those flatness anomalies above 300MHz in the CH Z could be  reconducible to the too thin traces of the bridge front-end, and in particular the thin trace which connects C12 to R13 pad. If I remember well, with a 1.6mm thick FR4 dielectric to get a 50ohm microstripline the width should be in the range of 3.9/4mm. 
Even if the FR4 pcb is surely not the best dielectric to use above 1GHz, but let me assert that we are not looking for a professional VNA, but the right compromise to give to the hobbyists a great low cost instrument.
In the figure that I attached below I did a mix  of the two LOG plots to evidence how much improvement we should get if the Z is fixed to 50ohm. The green area represent the spectral RL gain in terms of best matching.
BTW in case also the LO signal clock is driven with different impedances below and above the 300MHz point, it could be a a nice hint to check if the performance improves fixing that Z to 50ohm too (it could be not the case, because the signal it should be already sufficiently strong to saturate the mixers LO input).
Have a great day.

Massimo 


![50ohmImprovement](https://user-images.githubusercontent.com/57619411/69220124-914bd780-0b74-11ea-873b-20a17bdee7b0.jpg)

**Comment by qrp73** [comment]
> Shouldn't that impedance be left to 50ohm along the entire full 1500MHz span?

I'm also not sure if this impedance switch is really needed. Needs to try remove it and test results, may be it will give more dynamic range for 0-300 MHz range...

**Comment by qrp73** [comment]
just tried to use 50-ohm output for full frequency range, but it seems that it works a little worse. I didn't tried to play with gain, may be it can help to improve results.

I performed full calibration for 10 kHz - 500 MHz range, and take S11/S21 with 50-ohm loads on both ports.

First image is current version with impedance switch.
Second image is a version with si5351 constant 50-ohm output.

Current version with impedance switch at 300 MHz:
![si5351-switch](https://user-images.githubusercontent.com/46676744/69242883-7f852700-0baa-11ea-91be-fcf45a9af4ed.png)

Version with si5351 constant 50-ohm output:
![si5351-50](https://user-images.githubusercontent.com/46676744/69243060-e0acfa80-0baa-11ea-8b1c-9ee4d1c83650.png)

**Comment by Pmax65** [comment]
Hi qrp73,
effectively that impedance change is really a mess. As already I wrote to hugen, if this method was applied to the LO clock too, it could be a good hint to try set that output to 50ohm also below the 300MHz limit and see if it improves the performances of the instrument.
I hadn't been aware of it because I previously interpreted that "heartbeat" as a consequence of the nanoVNA outgoing signal, when you suggested the two spans test, I realized that the best (and fastest) way to get the Z measurement, it is setting the DUT to a CW frequency and then I clearly seen the impedance change. 
Thank you for having highlight me this issue which I was completely ignoring.

> First image is current version with impedance switch.
Second image is a version with si5351 constant 50-ohm output.

Do you have already made R13 = 0ohm; R9 = 200ohm and removed R29, R30 and R31?
Anyways it should be better to look at the full span to see if there are some issues on higher frequencies (I don't think, but it take a little to check it).

The RL seems to have lost dynamic, but having flatten a lot around the 10kHz level induces me to believe it's more realistic now, especially if the plot is less noisy now.

I don't know how noisy is that S21 plot, but it seems to me that the CH0/CH1 cross talk reduced a lot, is that or is it just an impression?

Massimo

**Comment by qrp73** [comment]
> Do you have already made R13 = 0ohm; R9 = 200ohm and removed R29, R30 and R31?

I tried it with default R13

> Anyways it should be better to look at the full span to see if there are some issues on higher frequencies

Frequencies higher than 300 MHz is not affected. Because current version uses 50-ohm output for frequencies above 300 MHz...

**Comment by Pmax65** [comment]
> I tried it with default R13

If you are saying that R13 is 150ohm then you can't get any improvement, because when the DUT changes amongst open-short-load, U6 isn't practically measuring the voltage drop at the bridge reference signal input to correct the measurements. Other than increasing the power level applied to the mixers LO inputs, that's the real improvement of the last mod that I proposed.

With R13 = 150 you can see how U6 is not working looking at the 300MHz and 900MHz bands cross points. Having still R13 = 150ohm, after a correct calibration, if you zoom around those cross point you can't see any spike due to the interpolation  between the 15MHz spaced reference points. In fact only one measurement is made exactly on that point. I guess that it is the one of the lower band of the twos. So it can't be used to compute the interpolated correction points for the upper band.
Making R13 = 0ohm you get the spikes that are expected because the interpolation point on the crossing band points lead to that wrong correction coefficients for the points between it and the first point of the higher band. For this reason, in those frequency ranges, I suggested to Hugen to change the calibration interpolation extending the last two points linearization of the lower band up to the crossband point and the first two points linearization of the upper band down to the crossband point.

> Frequencies higher than 300 MHz are not affected. Because current version uses 50-ohm output for frequencies above 300 MHz...

Yes, you are right. Effectively that check has no reasons to be done.

**Comment by qrp73** [comment]
the root of cause for spikes at 300, 600, ... MHz in the latest firmware is incorrect behavior of si5351 control code. This error is not hardware. It is completely software due to mistake. The driver for si5351 needs to be rewritten in order to fix that. 

But it's not so critical, because this issue happens when frequency point is 5 kHz away or more close to frequency boundary (300 / 600 / 900 / 1200 / 1500 MHz)

**Comment by Pmax65** [comment]
Hi qrp73.
No I'm not talking about that. The issue that arise making R13 = 0ohm is due to the now correct working of the reference mixer U6. The spikes you arguing can't be removed by a calibration, while the ones that I refer once you zoomed are completely removed by redoing the calibration with the stimulus set to the zoomed range.
Leaving R13 = 150ohm U6 measures the phase/amplitude of the clock generator output, not those of the weston bridge input port (the R10/R11 junction). This is last measuring point fundamental for getting the right RL plot. 
I don't know if this night I could do it, because I'm busy thit a RTX repair, but as soon I can do it I'll publish the simulation screenshots of what it happen when there is a load, a short and an open attached to the weston bridge DUT port and you'll can easily see that while the R10/R11 junction (the bridge unbalanced input port) changes a lot, the U6 picking point is practically unchanged.
Said that, there is a wrong behaviour of the interpolation algorithm when the interpolation is computed between the two interpolation reference points around the band switching point, or when the interpolation is computed between those two points and the very same  band switching point.
In the meantime, have a great day.

Massimo

**Comment by hugen79** [comment]
This is the schematic of my re-editing.
![Schematic_nanovna_Sheet_1_20191122131002](https://user-images.githubusercontent.com/32199654/69399622-cfd6c300-0d29-11ea-9e00-dde7f2f6f96d.png)

After calibration
![V3 4test](https://user-images.githubusercontent.com/32199654/69399664-ff85cb00-0d29-11ea-9731-4258ed4b6e31.png)

Measuring 20dB attenuator
![V3 4test_20dB](https://user-images.githubusercontent.com/32199654/69399701-1af0d600-0d2a-11ea-8bbd-aba2c9161742.png)

hugen

**Comment by Pmax65** [comment]
Hi hugen.
Well done!
Just to know:
A) why do your plots returned to have those peaks @ 300 & 900MHz?
B) In version 3.0 there was 3 100nF caps close to L3, L4 and L5 on the +5V side, aren't they no longer useful or it's an oversight? (I see that you named the new 100nF cap on the other side of the inductance with the same id)

I see you kept R29, R30 and R31 as not populated. Well done, I fully agree with your choice, because we should better investigate if overdriving the LO hasn't effectively unwanted side-effects.

By the way, if it's not too difficult to do for you, I would move the R16 terminal on the other side of R13. This way you could check if there are improvements in better adjusting the impedance that drive the input of the weston bridge (R10/R10 node), keeping the phase and amplitude monitoring by U6.
I don't want "teach the cat to climb the roof", but leave me suggest you to take care of the trace widths for the CLK2 signal and the CH0-CH1 ports.
Have a great day,

Massimo

**Comment by hugen79** [comment]
> Hi hugen.
> Well done!
> Just to know:
> A) why do your plots returned to have those peaks @ 300 & 900MHz?
> B) In version 3.0 there was 3 100nF caps close to L3, L4 and L5 on the +5V side, aren't they no longer useful or it's an oversight? (I see that you named the new 100nF cap on the other side of the inductance with the same id)
> 
> I see you kept R29, R30 and R31 as not populated. Well done, I fully agree with your choice, because we should better investigate if overdriving the LO hasn't effectively unwanted side-effects.
> 
> By the way, if it's not too difficult to do for you, I would move the R16 terminal on the other side of R13. This way you could check if there are improvements in better adjusting the impedance that drive the input of the weston bridge (R10/R10 node), keeping the phase and amplitude monitoring by U6.
> I don't want "teach the cat to climb the roof", but leave me suggest you to take care of the trace widths for the CLK2 signal and the CH0-CH1 ports.
> Have a great day,
> 
> Massimo


Hello Massimo, thank you for your suggestion.
The 300M and 900M anomalies come from the difference error of the calibration values, as discussed earlier, the harmonic switching point has a larger error. It cannot be solved by correcting the PGA gain, and this point must be separately calibrated, which requires further optimization of the software.
I originally thought that the capacitance on the +5V side could be omitted, but after testing today, they should not be omitted and have been added again. The position of R13 has moved, and edy555 also reminds me that I need to pay attention to the line width of CH0 and CH1. I try to increase the line width as much as possible.

hugen

![Schematic_nanovna_Sheet_1_20191122170138](https://user-images.githubusercontent.com/32199654/69414750-4be30200-0d4e-11ea-9172-b6edc8de65ab.png)

**Comment by Pmax65** [comment]
Hi Hugen,
just one other curiosity: I seen that you increased the CH1 input attenuator from 13 to 20dB. 
Does this change thought to improve the input impedance along the higher frequencies?

By the way, since the space in the board is not that narrow, couldn't you use 0603 components?
I'm getting older and it's so hard to me view those microscopic bugs!
(It's just joke, :-) of course) 

Have a great day.

Massimo

**Comment by hugen79** [comment]
Hello Massimo, after canceling R13, the output power of CH0 is increased by 10dB. If the input attenuation of CH1 is not increased, the signal input to CH1 will exceed the effective dynamic of U8.
In addition, I am preparing to further reduce the pcb area occupied by components. The larger area is used to install larger batteries. The next version will increase the battery to 650mAh, so I will still use 0402 components, although this is not friendly to DIYers.

hugen

**Comment by Pmax65** [comment]
Hi Hugen,
> after canceling R13, the output power of CH0 is increased by 10dB. If the input attenuation of CH1 is not increased, the signal input to CH1 will exceed the effective dynamic of U8.

yes, I didn't look at that (I had just looked at the CH1 background noise) and you are right.
ASAP, I'll change it as per your new schematic and I'll measure the new impedance on CH1 port. If he current higher frequencies mismatch was due to the mixer input capacitance it should be improved a little. But, honestòy, it's possible that it was due to the too thin (abt 1cm length) input trace. 
Once I'll measured it I tell you about.

> In addition, I am preparing to further reduce the pcb area occupied by components. The larger area is used to install larger batteries. The next version will increase the battery to 650mAh, so I will still use 0402 components, although this is not friendly to DIYers.

As said, I was joking of course :-D
I wrote that just to complain with myself about my reduced capability of see these microscopic components of today. Unluckily, age has its pros & cons :-(

Have a great day.

Massimo

**Comment by Pmax65** [comment]
Hi qrp73,
as promised, I publish here below the simulations about the behaviour of the signal at reference mixer U6 with R13 = 150ohm and with R13 = 0ohm.
As you can see WITH r13 = 150ohm, despite the impedance seen on the DUT port forces a wide signal variation on the bridge input port (the joint between R10 and R11, which change from 275mV to 417mV), the signal variation on the reference mixer input is tiny (126mV to 128mV).

![Old Reference](https://user-images.githubusercontent.com/57619411/69493092-b315c900-0eaa-11ea-90e6-b6fc1f0583cd.jpg)

Instead using R13 = 0ohm the changes on the bridge input (R10/R11 joint, which change from 924mV to 1.24V) are well reflected at the reference mixer input (66.4mV to 49.4mV).

![New Reference](https://user-images.githubusercontent.com/57619411/69493095-b446f600-0eaa-11ea-9449-9c6bac06a09a.jpg)

Have a great day.

Massimo

**Comment by Pmax65** [comment]
Hi Hugen.
I did the measurements on the CH1 port and doing that I also discovered that the input measurements that I did 4 days ago on CH0 by night were wrong (my neurons were probably sleeping that time). 
I did them having calibrated the HP8711A still too cold, not waiting for its warm up, and I didn't notice that those ondulation seen above 300MHz were due to the instrument drift.
This is the true CH0 Zin:
![Ch0 Zin](https://user-images.githubusercontent.com/57619411/69493293-78f9f680-0ead-11ea-81e0-f54d5f163968.jpg)
As you can see, it's much better the those I published 4 days ago indeed.
Those more than -23dB of return loss make the instrument uncertainty lesser than +/-0.67dB@1.3GHz and those more than -29dB of return loss at 900MHz make the instrument uncertainty lesser than +/-0.21dB that I think is gorgeous for this very cheap device.

About V3.4.2 CH1 Zin, as expected, it improves even if only a little:
![CH1 Z](https://user-images.githubusercontent.com/57619411/69493611-f07d5500-0eb0-11ea-9f30-c2cf743f6219.jpg)
Note how the little residual ondulation of the return loss it could be due to the fact that I don't have references open, short and load having an SMA-M connector, so I had to use a small SMA M-M adapter inserted between the HP0711A S11 calibrated port and the DUT ports. The male to male adapter reference planes distance it was about 16mm and that could be the culprit for that small ondulation.

Have a great day.

Massimo

**Comment by Pmax65** [comment]
Hi Hugen,
I missed to tell you that the new CH1 impedance I published above is for R22 and R23 = 51 ohm not 49.9ohm since I don't have that value in my sample book.
The 51ohm resistors should behave a little better because the simulated impedance comprising the mixer input impedance is 49.59ohm, while the 49.9ohm resistors lead to a CH1 Zin of 49.16ohm.

Have a great day.

Massimo

**Comment by hugen79** [comment]
> Hi Hugen,
> I missed to tell you that the new CH1 impedance I published above is for R22 and R23 = 51 ohm not 49.9ohm since I don't have that value in my sample book.
> The 51ohm resistors should behave a little better because the simulated impedance comprising the mixer input impedance is 49.59ohm, while the 49.9ohm resistors lead to a CH1 Zin of 49.16ohm.
> 
> Have a great day.
> 
> Massimo



You are right, I will update, thank you!

hugen

**Comment by hugen79** [comment]
> Hi Hugen,
> I missed to tell you that the new CH1 impedance I published above is for R22 and R23 = 51 ohm not 49.9ohm since I don't have that value in my sample book.
> The 51ohm resistors should behave a little better because the simulated impedance comprising the mixer input impedance is 49.59ohm, while the 49.9ohm resistors lead to a CH1 Zin of 49.16ohm.
> 
> Have a great day.
> 
> Massimo

Hello Pmax65, I have already made a PCB for V3.4, and send your shipping address to my email: hugen@outlook.com, and I will send you a new version of NanoVNA-H.

hugen

**Comment by hugen79** [comment]
> the root of cause for spikes at 300, 600, ... MHz in the latest firmware is incorrect behavior of si5351 control code. This error is not hardware. It is completely software due to mistake. The driver for si5351 needs to be rewritten in order to fix that.
> 
> But it's not so critical, because this issue happens when frequency point is 5 kHz away or more close to frequency boundary (300 / 600 / 900 / 1200 / 1500 MHz)

Hello Qrp73, I have already made a PCB for V3.4, and send your shipping address to my email: hugen@outlook.com, and I will send you a new version of NanoVNA-H.

hugen

**Comment by Pmax65** [comment]
Hi Hugen,

I just realized that you sent the message below to me instead of sending it to 
Qrp73.

Check if you already sent it to him too

---
New Outlook Express and Windows Live Mail replacement - get it here:
https://www.oeclassic.com/

.

Have a great day.

Massimo





----- Original Message -----

**Comment by Pmax65** [comment]
Hi Hugen,
I discovered a new issue that probably could be solved (or at least reduced) 
via firmware modification.
Attached to this message you can see an image of the analysis of the VHF port 
of one of my VHF/UHF duplexer. 
The black plots are from my HP8711A VNA, while the red plot is the nanoVNA S21 
and the blue plot is the nanoVNa S11.
 
While the S11 mismatching 
isn't a real issue because my HP8711A was calibrated using a N connectors calibration kit, and that easily justify the discrepancy at the lower frequency.
The S21 has a clear problem at the 300MHz band-switch crossover point.
 
Analyzing the spectrum of the nanoVNA CH0 output set to a CW frequency at 310, 312 and 315MHz, I discovered a series of sub-carriers around the 3rd harmonic fundamentals at respectively 103.333MHz, 104MHz and 105MHz which could justify that noise excess between 300 and 600MHz because those sub-carriers have themselves sub-carriers shifted exactly 5kHz from their carrier which fit the nanoVNA IF.
Since those sub-carrier are shifted from the fundamental of a frequency that it is equal to the difference of the 5th harmonics fundamental and a multiple of 8MHz which is the MCU clock, I tried to smooth the 8MHz output with a capacitor, but despite it changed from a square-wave to a triangle one, the issue doesn't 
improve. It improved, matching the HP8711A between 300MHz and 384MHz, when I 
inserted a 10dB attenuator on the CH1 input (probably because the +3dBm CH0 
output is still a little too high for the CH1 3rd IP), but there still was a 
Gaussian shaped noise between 384MHz and 528MHz that peaked at 450MHz.
 
I suspect that this is a cross-talk of the outputs buffers inside the Si5351. 
Since the Si5351 allow the spread spectrum on any single output, I would like 
to know if you could compile a special version of your firmware with the spread 
spectrum activated on output 2 set to the maximum spread (+/-1.5% in center 
spread mode). 
 
It's just a check, so it shouldn't be a new release of the firmware for the 
moment.
 
Have a great day.
 
Massimo

---
New Outlook Express and Windows Live Mail replacement - get it here:
https://www.oeclassic.com/

**Comment by Pmax65** [comment]
Hoops!
Sorry I missed to attache the image to my previous message.

---
New Outlook Express and Windows Live Mail replacement - get it here:
https://www.oeclassic.com/

**Comment by Pmax65** [comment]
I missed one more point (sorry).
In case you can supply me a version of the FW with the spread spectrum enabled 
on output 2, and the ouput current wasn't already set to the minimum, it would be better if the 8MHz output 2 was set to the minimum current (2mA) of course.

---
New Outlook Express and Windows Live Mail replacement - get it here:
https://www.oeclassic.com/


Massimo

**Comment by hugen79** [comment]
> I missed one more point (sorry). In case you can supply me a version of the FW with the spread spectrum enabled on output 2, and the ouput current wasn't already set to the minimum, it would be better if the 8MHz output 2 was set to the minimum current (2mA) of course.
> […](#)
> --- New Outlook Express and Windows Live Mail replacement - get it here: https://www.oeclassic.com/ Massimo

Sorry, I still don't see the picture, you can use the "power 0" command to set the output power of the si5351 to 2mA.

hugen

**Comment by Pmax65** [comment]
Hi Hugen,
I sent the picture again attached to this message in pdf format.

Your wrote "you can use the "power 
0" command to set the output power of the si5351 to 2mA."
I need to reduce the current only from from Si5351 output 2, which is the 8MHz 
MCU clock, is that command available for output 2 too?

Anyways I gone ahead and I reduced that noise down to -40dB from the carrier 
just installing the 150 ohms resistor (ex R13) in place of C12, and in place of 
R13 I had put a 10uF X7R  capacitor, doing that the cross-talk reduced a lot. 
It seems to be a know issue for the Si5351 I found some people arguing about 
that into an Arduino SDR blog. It seems that it is all due to the maximum 
current allowed to the chip, in case it exceed 5.6mA per output a cross-talk 
happens in the form of jitter on the output signal.

Have a great day.

Massimo

---
New Outlook Express and Windows Live Mail replacement - get it here:
https://www.oeclassic.com/





----- Original Message -----

**Comment by Pmax65** [comment]
Hi Hugen,
I feel like an idiot!
Since you sent me also private emails, I was not aware that this was the GitHub  email and replying to that address, evidently the email server removes the attachments.
Here is the image I was referring to:
![Test 300kHz-600MHz](https://user-images.githubusercontent.com/57619411/71615091-bb4edb80-2baf-11ea-9676-30b03b44a5cc.jpg)

After changing C12 to 150ohms and R13 to 10uF the spurious noise fallen by 10dB.
Now the output signal from CH0 is what I can get with the old original configuration with R13 = 150ohm, but now the reference mixer U6 still works correctly.
Note that in your new PCB having moved R13 before the reference picking point, it suffices to make R13=150ohms again without the need of the 10uF capacitor.  
Today I'm busy and I don't know if I could do other tests, but next I'll try other checks to see if I can improve its behaviour more.

Happy New Year 

Massimo

**Comment by hugen79** [comment]
Happy new year Pmax65, I made a simple low-pass filter for this test and got some similar results. This is really a bad problem and needs to be addressed in the next release. Our previous improvement increased the output power of ch0, but the noise also became larger. I will try your suggestions for further improvement.

No matter what the state is, the output 2 of si5351 in nanovna is 2mA.

> Hi Hugen,
> I feel like an idiot!
> Since you sent me also private emails, I was not aware that this was the GitHub email and replying to that address, evidently the email server removes the attachments.
> Here is the image I was referring to:
> ![Test 300kHz-600MHz](https://user-images.githubusercontent.com/57619411/71615091-bb4edb80-2baf-11ea-9676-30b03b44a5cc.jpg)
> 
> After changing C12 to 150ohms and R13 to 10uF the spurious noise fallen by 10dB.
> Now the output signal from CH0 is what I can get with the old original configuration with R13 = 150ohm, but now the reference mixer U6 still works correctly.
> Note that in your new PCB having moved R13 before the reference picking point, it suffices to make R13=150ohms again without the need of the 10uF capacitor.
> Today I'm busy and I don't know if I could do other tests, but next I'll try other checks to see if I can improve its behaviour more.
> 
> Happy New Year
> 
> Massimo


Happy new year Pmax65, I made a simple low-pass filter for this test and got some similar results. This is really a bad problem and needs to be addressed in the next release. Our previous improvement increased the output power of ch0, but the noise also became larger. I will try your suggestions for further improvement.

No matter what the state is, the output 2 of si5351 in nanovna is 2mA.

hugen

**Comment by Pmax65** [comment]
Hi Hugen,
I had some improvements placing a 10pF capacitor piggyback to the 150ohm resistor. 
In this case, between 300 and 900MHz the noise remains below the -60dB level and when the DUT is a low-pass filter that allow the fundamental to pass through and reach the CH1 port, the noise improves of about 10dB, but the dynamic of CH1 between 100MHz and 300MHz deteriorates in the range from 70dB to 55dB.

I would like to see if enabling the spread spectrum on the Si5351 output 2 the situation improves or not.
By the way, the phase noise on that 8MHz output is already terrific because of the LED2 blinking, 
For that, enabling the spread spectrum shouldn't be an issue for the MCU, it should reduces the global noise generated by the instrument instead.

Happy New Year again,

Massimo

**Comment by hugen79** [comment]
Hello Massimo, Only Si5351's PLLA allows the use of spread spectrum, so I have no way to enable the spread spectrum of CLK2. But my new NanoVNA-H4 is equipped with a separate 8M crystal resonator, so I can turn off the CLK2 of Si5351 on my new NanoVNA-H4, but the measurement results have not improved.

hugen

**Comment by Pmax65** [comment]
Hi Hugen.
Ok, that's was the next check I would do (that is, use a separate clock for the MCU), but since you already did it. It seems that it is something about the cross-talk between CH0 and CH1 at this point.  
Anyways, it is surely a signal embedded in the CH0 output signal, because the CH1 terminated on a dummy load hasn't that noise there. For that, it shouldn't be embed in the mixers LO signals.

In the past month I already did some tests matching the mixers input Z for the best power transfer, but unexpectedly I got less dynamic from the CH1 port. In fact I had a lot of non-thermal noisy background along the whole 10kHz/1.5GHz spectrum. Probably the mixers went more sensitive receiving the various clock harmonics of the MCU and DSP whic probably are "unlistened" with the mixer input Z mismatched.  

Honestly the nanoVNA is a great instrument anyway as it is. All my HAM-radio friend already get back their modified one and had experienced positive results using it for antenna tuning, which was the main reason they asked me to get it working at the 23cm band, where it was very noisy before.

Massimo



Massimo

**Comment by Pmax65** [comment]
Hi Hugen,
first of all thank you for the firmware updates, I'll give them a try next.
Instead, yesterday I gave a try to the new PCBs based instruments and (as already said), they are better than the older one.
Anyways returning the now correctly placed R13 to 150 ohm and making:
R25 = 22 ohm
R23 = 0 ohm 
R22 = 68 ohm
R24 = 120 ohm
The CH0 and specially the CH1 impedances are better matched to the Z0 = 50 ohm.
Now the CH0 return loss is better than -20dB for the entire 10kHz-1.3GHz span and the CH1 return loss is better than a gorgeous -30dB up to 1GHz and -27dB up to 1.3GHz.
Note that I don't have a VNA up to 1.5GHz to check the return loss there, but using an HP86205A directional bridge with my Advantest R3131A spectrum analyzer and its tracking generator, the CH1 return loss looks better than -30dB along the whole 10kHz-1.3GHz (this tells all about the low accuracy on that low return loss levels).

For this reason I suggest you to change those components in the next production batch,

Returning instead to the low pass filter in baseband with stop band in the harmonics bands issue: I see that in nanoVNA-H4 you placed R49 and R50 to select the crystal or the Si5351, but I have a doubt about your previous check.
Do you just disabled the output amplifier of the Si5351 channel 2?
Reading a little more about that clock generator, it has a multiplexer matrix to select the output amplifiers sources and the internal dividers, I suspect that the interferences between channels could be due to their cross-talks, and not to the device ground quality as suggested from the people of the Arduino based SDR. 
If I'm right, disabling the output amplifier doesn't do anything regarding this issue.
You should try to drive the MCU and the DSP with the crystal and set the channel 2 to the very same value of channel 1, since the Si5351 is made to supply clock of the very same frequency at different outputs just phase-shifted, in case I'm right all those subcarriers should clear away.
If you are busy, I could do it this check. Could you just give me a link for a Windows7 Compiler for those MCUs?

Have a great day.

Massimo

**Comment by hugen79** [comment]
Hello Massimo, if you restore R13 to 150 ohms, CH0 cannot measure to 1.5GHz. I have carried out some new tests on the impedance of CH1, and I will also perform new tests as you suggest.

You can refer to this link to install the IDE, pay attention to using GCC 8.
https://gnu-mcu-eclipse.github.io/downloads/

Comment out line 30 of nanovna.h of the F303 branch to turn off the clk2 output of NanoVNA-H4.

hugen

**Comment by Pmax65** [comment]
Hi Hugen,
thank you for the link to the IDE.

> if you restore R13 to 150 ohms, CH0 cannot measure to 1.5GHz.

Really? I was talking with R13 placed between C12 and R14 as per the 3.4 schematic of course.
At this point, if you were right, it could be that I did something wrong in the check. 
I write this because I'm almost sure that checking the 33 ohms and the 75 ohms 1:1.5 SWR references resistors, I had straight -13dB return loss lines along the whole 10kHz-1.5GHz range and two small lines positioned in the Smith's chart close to the resistive axis very close to the resistor resistance value (if I remember well there was few pH for the 33ohm reference and few pF for the 75ohm reference, probably due to the cable not well compensated above 1GHz). 
Let me check it again and I tell you something.

Anyway, I don't suggest to modify those resistors until the next production batch, because the current advantage is of few dB in the range between 300 and 380MHz.

Have a great day.

Massimo

**Comment by d9thc31** [comment]
Hi Hugen, please tell me where you can see the last version of the circuit - REV: 3.4?
I want to bring my REV: 3.3 as close as possible to the REV: 3.4 version without changing the circuit board.
And please tell me what you think about the use of resistors R22 and R23 with a resistance of 51 Ohms instead of 49.9 Ohms, unfortunately I can’t get 49.9 Ohms resistors right now.

I apologize if I am not accurate in words, I live in Russia and unfortunately do not know English, so I have to use a translator.

**Comment by Pmax65** [comment]
Hi Dmitry,
the 51ohm resistors behave better than the 49.9ohm for the port mach indeed. 
Look at this link: https://github.com/hugen79/NanoVNA-H/issues/14#issuecomment-557908195

Attached to this message you can find my docs to apply the modifications.
Please carefully read the warning in the frontpage.

Revision 1.0 is for make versions before 3.4 more or less similar to 3.4 (the pcb and the change of the dc-dc converter in version 3.4 make it different, but the performance are very little better).
Revision 1.1 is to improve the dynamic of CH1 when the DUT is a low-pass filter in the fundamental 10k-300MHz band. This degrades a little the S11 dynamic, but make useful the S21 in the range 300-400MHz (who really cares to get better than 60dB of return loss measurements?). 

[NanoVNA Rev. 3.0 Mods.1.0.pdf](https://github.com/hugen79/NanoVNA-H/files/4142428/NanoVNA.Rev.3.0.Mods.1.0.pdf)
[NanoVNA Rev. 3.0 Mods.1.1.pdf](https://github.com/hugen79/NanoVNA-H/files/4142429/NanoVNA.Rev.3.0.Mods.1.1.pdf)

Have a great day.

Massimo IK1IZA

**Comment by d9thc31** [comment]
> Hi Dmitry,
> the 51ohm resistors behave better than the 49.9ohm for the port mach indeed.
> Look at this link: [#14 (comment)](https://github.com/hugen79/NanoVNA-H/issues/14#issuecomment-557908195)
> 
> Attached to this message you can find my docs to apply the modifications.
> Please carefully read the warning in the frontpage.
> 
> Revision 1.0 is for make versions before 3.4 more or less similar to 3.4 (the pcb and the change of the dc-dc converter in version 3.4 make it different, but the performance are very little better).
> Revision 1.1 is to improve the dynamic of CH1 when the DUT is a low-pass filter in the fundamental 10k-300MHz band. This degrades a little the S11 dynamic, but make useful the S21 in the range 300-400MHz (who really cares to get better than 60dB of return loss measurements?).
> 
> [NanoVNA Rev. 3.0 Mods.1.0.pdf](https://github.com/hugen79/NanoVNA-H/files/4142428/NanoVNA.Rev.3.0.Mods.1.0.pdf)
> [NanoVNA Rev. 3.0 Mods.1.1.pdf](https://github.com/hugen79/NanoVNA-H/files/4142434/NanoVNA.Rev.3.0.Mods.1.1.pdf)
> 
> Have a great day.
> 
> Massimo IK1IZA

Hi Massimo, thanks for helping me, I am happy to follow your experiments to improve the analyzer.
Have a nice day!

Dmitriy
